pkg/cloudprovider/aws/fleet/launchtemplate.go

```diff
@@ -36,6 +37,14 @@ import (
 const (
 	LaunchTemplateNameFormat = "Karpenter-%s"
 	IAMInstanceProfileName   = "KarpenterNodeRole"
+	bottleRocketUserData     = `
+[settings.kubernetes]
+api-server = "${.Endpoint}"
+cluster-certificate = "${.CABundle}"
+cluster-name = "${.Name}"
+[settings.kubernetes.node-labels]
```

Comment:
Check out the labels we're applying with our controller: https://github.com/awslabs/karpenter/blob/main/pkg/controllers/provisioning/v1alpha1/allocator/bind.go#L56. Probably still a good idea to add some label, in case our controller crashes. We want to be able to avoid the leakage case.

Reply:
I think I'm confused about that - I was just copying what the UserData was for the EKS Optimized AMI case; are you saying that the additional labels should be moved here too?

Reply:
These two are different labels, this one here is applied to the EC2 instance and the one in `bind.go` gets applied to the node object in the cluster. 
When we start supporting multiple Karpenter instances per cluster/account, at that time we might want to add the same label(s) here that we add to node object in Kubernetes. This will help identify which Karpenter instance created this EC2 VM.

Reply:
Hmm.. now I'm wondering if the above `settings...` thing is working:

I'm not seeing the label either as an EC2 tag or as a k8s label on the node ...

```
  labels:
    beta.kubernetes.io/arch: amd64
    beta.kubernetes.io/instance-type: m5.large
    beta.kubernetes.io/os: linux
    failure-domain.beta.kubernetes.io/region: us-west-2
    failure-domain.beta.kubernetes.io/zone: us-west-2a
    kubernetes.io/arch: amd64
    kubernetes.io/hostname: ip-192-168-131-169.us-west-2.compute.internal
    kubernetes.io/os: linux
    node.kubernetes.io/instance-type: m5.large
    provisioning.karpenter.sh/name: default
    provisioning.karpenter.sh/namespace: default
    topology.kubernetes.io/region: us-west-2
    topology.kubernetes.io/zone: us-west-2a
```

Reply:
Our controller will apply many labels to the node. In this context, it's probably only necessary to add in the name/namespace ones so we at least know ownership. The other labels are a bit complex, and shouldn't be on a per-launch-template basis.

It's also a little tricky because if we want to add name/namespace labels here, then we'll need one launch template per provisioning group. That's a nontrivial code change.

If you're happy to make it, that's great. Like wire through the provisioner and change our cache key to provisioner.Name/provisioner.Namespace. Feel free to punt to a later date, though.

Reply:
I think I'm still missing a basic question, why did the old code do this? Are you saying it was unnecessary?

```
--kubelet-extra-args '--node-labels=karpenter.sh/provisioned=true' \
```

Reply:
Oh that was a total accident :)